lib/rules/no-test-support-import.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileOverview no importing of test files and no exporting in test files
+ */
+
+'use strict';
+
+const ERROR_MESSAGE_NO_IMPORT =
+  'Do not import a file from test-support into production code, only into test files.';
+
+module.exports = {
+  ERROR_MESSAGE_NO_IMPORT,
+
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow importing of "test-support" files in production code.',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-test-support-import.md',
+    },
+    schema: [],
+  },
+
+  create: function create(context) {
+    const fileName = context.getFilename();
+
+    return {
+      ImportDeclaration(node) {
+        const importSource = node.source.value;
+
+        if (
+          importSource.includes('test-support') &&
```

Comment:
Can we improve this to be sure it only catches exact directory name matches? And add a test case? We don't want this to catch partial matches like `my/test-support-foo/path`. 

```js
function hasTestSupportDirectory(path) {
  return (
    importSource.startsWith('test-support/') || 
    importSource.endsWith('/test-support') ||
    importSource.includes('/test-support/')
  );
}
```

Reply:
added this.

Reply:
Can you add a test case with a `/test-support-foo/` import?

Reply:
add a new test to ensure this case would fail.